hs_core/views/utils.py

```diff
@@ -258,3 +266,42 @@ def get_my_resources_list(request):
 
     resource_collection = (owned_resources + editable_resources + viewable_resources + discovered_resources)
     return resource_collection
+
+def send_action_to_take_email(request, user, action_type, **kwargs):
+    """
+    Sends an email with an action link to a user.
+    The actual action takes place when the user clicks on the link
+
+    The ``action_type`` arg is both the name of the urlpattern for
+    the action link, as well as the names of the email templates
+    to use. Additional context variable needed in the email template can be
+    passed using the kwargs
+
+    for action_type == 'group_membership', an instance of GroupMembershipRequest and instance of Group are expected to
+    be passed into this function
+    """
+    email_to = kwargs.get('group_owner', user)
+    context = {'request': request, 'user':user}
+    if action_type == 'group_membership':
+        membership_request = kwargs['membership_request']
+        action_url = reverse(action_type, kwargs={
+            "uidb36": int_to_base36(email_to.id),
+            "token": default_token_generator.make_token(email_to),
+            "membership_request_id": membership_request.id
+        }) + "?next=" + (next_url(request) or "/")
+
+        context['group'] = kwargs.pop('group')
+    else:
+        action_url = reverse(action_type, kwargs={
+            "uidb36": int_to_base36(email_to.id),
+            "token": default_token_generator.make_token(email_to)
+        }) + "?next=" + (next_url(request) or "/")
+
+    context['action_url'] = action_url
+    context.update(kwargs)
+
+    subject_template_name = "email/%s_subject.txt" % action_type
+    subject = subject_template(subject_template_name, context)
+    send_mail_template(subject, "email/%s" % action_type,
+                       settings.DEFAULT_FROM_EMAIL, email_to.email,
+                       context=context, fail_silently=True)
```

Comment:
@pkdash Same as above, I think in this case, fail_silently should be set to False, which is the default, so you can just remove this fail_silently assignment in send_mail_template() call.